src/main/java/com/launchdarkly/client/LDUser.java

```diff
@@ -143,6 +147,32 @@ JsonElement getCustom(String key) {
     return null;
   }
 
+  JsonObject toJsConfig(LDConfig config) {
+    JsonElement tree = new Gson().toJsonTree(this);
+
+    Set<String> privateAttributeSet = new HashSet<>();
+    privateAttributeSet.addAll(this.privateAttributeNames);
+    privateAttributeSet.addAll(config.privateAttrNames);
```

Comment:
I'm not sure of the reason for this line. The front end already has a `privateAttributeNames` configuration option for the client, just like the back end; if you're using that option on the back end, presumably you would be doing the same on the front end. The `privateAttributeNames` property in the user is meant for attributes that have been marked private for _this user_.

I mean, doing this won't hurt anything, but I think it is unnecessary and it also conflicts with what is arguably the normal goal of JSON serialization: capturing the state of this particular object.

Reply:
Similarly, I think adding the `allAttributesPrivate` property from the client configuration into the user object is misleading. If you want your front-end client to always use the same configuration as the back end without having to specify it in code, I think it would be better to pass those configuration options separately.